cloudapi/logs.go

```diff
@@ -166,6 +173,18 @@ func (c *Config) StreamLogsToLogger(
 		if err != nil {
 			logger.WithError(err).Warn("error reading a message from the cloud")
 
+			// try to restore the stream establishing a new connection
+			logger.Warn("trying to establish a fresh connection with the tail logs, this might result in either some repeated or missed messages") //nolint:lll
+
+			newconn, errd := c.logtailConn(ctx, referenceID, time.Now())
```

Comment:
given how time can drift due to network latency and be different between the client and the server, we probably should use the time of the last received log message (`+1ms` to avoid repetition) here. We can only fall back to `time.Now()` if we haven't received any log messages so far